### ISSUE TITLE
global name 'template' not defined

### ISSUE THREAD

**Issue by wsmorgan** [rosenbrockc/fortpy#80]
When a template is specified in the .xml for a test output, ie:
```
	<output folder="../tests/derivative_structure_generator" file="struct_enum.out.{}"
		template="struct_enum.out.xml" identifier="struct_enum.out" />
```
fortpy returns this error after running the tests before doing any comparisons:
```
NameError: global name 'template' is not defined
```

**Comment by wsmorgan** [comment]
The full error backtrace.

```
Traceback (most recent call last):
  File "/Users/wileymorgan/.virtualenvs/fortpy/bin/runtests.py", line 147, in <module>
    do_testing()
  File "/Users/wileymorgan/.virtualenvs/fortpy/bin/runtests.py", line 49, in do_testing
    result = t.runall(c)
  File "/Users/wileymorgan/.virtualenvs/fortpy/lib/python2.7/site-packages/fortpy/testing/tester.py", line 617, in runall
    oneresult = self._run_single(identifier, testid, source)
  File "/Users/wileymorgan/.virtualenvs/fortpy/lib/python2.7/site-packages/fortpy/testing/tester.py", line 633, in _run_single
    self._run_exec(identifier, testid, result)
  File "/Users/wileymorgan/.virtualenvs/fortpy/lib/python2.7/site-packages/fortpy/testing/tester.py", line 708, in _run_exec
    xres.test(case, result)
  File "/Users/wileymorgan/.virtualenvs/fortpy/lib/python2.7/site-packages/fortpy/testing/tester.py", line 61, in test
    self.tester.test(caseid, self, uresult)
  File "/Users/wileymorgan/.virtualenvs/fortpy/lib/python2.7/site-packages/fortpy/testing/tester.py", line 135, in test
    result = self._run_compare_file(exepath, outvar, self.codefolder, caseid)
  File "/Users/wileymorgan/.virtualenvs/fortpy/lib/python2.7/site-packages/fortpy/testing/tester.py", line 269, in _run_compare_file
    result = self.comparer.compare(exepath, targetpath, outvar.template, outvar.mode)
  File "/Users/wileymorgan/.virtualenvs/fortpy/lib/python2.7/site-packages/fortpy/testing/comparer.py", line 505, in compare
    result = compare_representations(svalues, tvalues, mode)
  File "/Users/wileymorgan/.virtualenvs/fortpy/lib/python2.7/site-packages/fortpy/testing/comparer.py", line 216, in compare_representations
    body = _compare_body(rep1, rep2, mode)
  File "/Users/wileymorgan/.virtualenvs/fortpy/lib/python2.7/site-packages/fortpy/testing/comparer.py", line 264, in _compare_body
    thisd = _get_keyed_dict(rep1.template, rep1.body)
  File "/Users/wileymorgan/.virtualenvs/fortpy/lib/python2.7/site-packages/fortpy/testing/comparer.py", line 414, in _get_keyed_dict
    keyval = _get_key_value_list(template, block)
  File "/Users/wileymorgan/.virtualenvs/fortpy/lib/python2.7/site-packages/fortpy/testing/comparer.py", line 425, in _get_key_value_list
    values = [_get_key_value_single(k, bodyblock) for k in template.key]
  File "/Users/wileymorgan/.virtualenvs/fortpy/lib/python2.7/site-packages/fortpy/testing/comparer.py", line 458, in _get_key_value_single
    msg.err("block {} did not return a valid key value for '{}'.".format(bodyblock, template.key))
```

**Comment by rosenbrockc** [comment]
Fixed in Revision 1.7.3
